src/library/styles/__tests__/createStyledComponent.spec.js

```diff
@@ -0,0 +1,121 @@
+/* @flow */
+import React from 'react';
+import { mount } from 'enzyme';
+import { createStyledComponent } from '../../styles';
+
+const mountButton = (props = {}, styles = {}, options = {}) => {
+  const Button = createStyledComponent('button', styles, options);
+  return mount(<Button {...props} />);
+};
+
+describe('createStyledComponent', () => {
+  const wrapper = mountButton({ type: 'button', xyz: true });
+  const button = wrapper.childAt(0);
+
+  it('forwards valid DOM attributes', () => {
+    expect(button.props().type).toBeDefined();
+  });
+
+  it('filters invalid DOM attributes', () => {
+    expect(button.props().xyz).toBeUndefined();
+  });
+
+  describe('with withProps option', () => {
+    const wrapper = mountButton(
+      {},
+      {},
+      {
+        withProps: {
+          type: 'button'
+        }
+      }
+    );
+    const button = wrapper.childAt(0);
+
+    it('passes props to component', () => {
+      expect(button.props().type).toBe('button');
+    });
+  });
+
+  describe('with filterProps option', () => {
+    const wrapper = mountButton(
+      { filterMe: true },
+      {},
+      {
+        filterProps: ['filterMe']
+      }
+    );
+    const button = wrapper.childAt(0);
+
+    it('filters filterProps', () => {
+      expect(button.props().filterMe).toBeUndefined();
+    });
+  });
+
+  describe('with forwardProps option', () => {
+    const wrapper = mountButton(
+      { forwardMe: true },
+      {},
+      {
+        forwardProps: ['forwardMe']
+      }
+    );
+    const button = wrapper.childAt(0);
+
+    it('forwards forwardMe', () => {
+      expect(button.props().forwardMe).toBeDefined();
```

Comment:
I may be misunderstanding, but I don't understand how this assertion is meaningfully different from the one on line 16. Wouldn't you need to forward a non-valid DOM attribute for this to be useful? Or, put another way, is asserting on `.props()` an accurate representation of what's actually in the DOM?

Reply:
`forwardMe` is an invalid DOM attribute.  Line 16 asserts on a valid DOM attribute.

Regarding it being valid to test `.props()`, that is a good question.  I think so since, commenting out the forwardProps option causes the test to fail.  FWIW, I lowercased the forwardme prop and set the value to a string, which gets rid of the console warning and also causes the attribute to show when logging button.html().  I could also assert on the button.html() string, but not sure if it is worth it as this seems to already be testing the mineral part of forwardProps functionality.  I think a snapshot would essentially be the same as it is currently, just less explicit.  Open to other suggestions…